src/output/shell_output.go

```diff
@@ -69,6 +69,16 @@ func MonitorState(state *core.BuildState, numThreads int, plainOutput, keepGoing
 	failedTargets := []core.BuildLabel{}
 	failedNonTests := []core.BuildLabel{}
 	for result := range state.Results {
+		if state.DebugTests && result.Status == core.TargetTesting {
+			stop <- struct{}{}
+			<-displayDone
+			// Ensure that this works again later and we don't deadlock
+			// TODO(peterebden): this does not seem like a gloriously elegant synchronisation mechanism...
+			go func() {
+				<-stop
+				displayDone <- struct{}{}
+			}()
```

Comment:
Hmmmm, not sure that is right; `logProgress` is only called if the interactive display isn't running, and it would do its thing asynchronously which isn't what I want here.
I agree it is messy though, as my comment indicates, but I think I'd rather rethink this as a synchronisation mechanism in a separate PR; it's a very old piece of code and it's possible the author didn't know Go so well at the time he wrote it 😇

Reply:
Hehe, ok, fair :)